services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,38 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
-
+  if (xQueueSend(thermalMgrQueueHandle, event, (TickType_t) 0) != pdPASS) {
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event = { .type = OS_HANDLER_CHECK_OVER_TEMP };
+  xQueueSend(thermalMgrQueueHandle, &event, (TickType_t) 0);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  thermal_mgr_event_t buf;
+  float temp;
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &(buf), ( TickType_t ) 10) == pdTRUE) {
```

Comment:
Use a block time of portMAX_DELAY as it will block the task until an event is received in the queue